hcldec/spec.go

```diff
@@ -455,13 +461,23 @@ func (s *BlockListSpec) decode(content *hcl.BodyContent, blockLabels []blockLabe
 		panic("BlockListSpec with no Nested Spec")
 	}
 
+	// If any block Body is unknown, then the entire block value must be
+	// unknown
+	unknown := false
+
 	var elems []cty.Value
 	var sourceRanges []hcl.Range
 	for _, childBlock := range content.Blocks {
 		if childBlock.Type != s.TypeName {
 			continue
 		}
 
+		if u, ok := childBlock.Body.(UnknownBody); ok {
+			if unknown = u.Unknown(); unknown {
```

Comment:
Perhaps I'm not reading this right, but this seems like it would get an incorrect result if there was an unknown block followed by a known one, because the second one would reset `unknown` back to `false` again.

I guess this doesn't arise in practice right now because there isn't currently any situation where you'd have two `UnknownBody` implementations in the mix that disagree on the `Unknown` result, but since this is an exported interface there could be other implementations of it in future that decide the result differently, and thus lead to a mixture of `true` and `false` results in the same loop. I think... right? :thinking:

Reply:
We `break` immediately on the first true `unknown`, so the value doesn't have a chance to be overwritten by a later known body. Once we know that we are gong down the unknown path, we don't need to process any more elements.

Reply:
In reviewing the code to coddle check this, I see we can use this to fix the min/max validations now too. Terraform needs to overwrite those in the spec to prevent false positives with dynamic, preventing any useful size validation at all.

Reply:
Ahh yes, with fresh eyes today I see that I must've been somehow misreading `break` as `continue` when I was reviewing this yesterday. Sorry for the misdirection.